### ISSUE TITLE
Item with same id already exists  

### ISSUE THREAD

**Issue by gotgameio** [maxnowack/signaldb#607]
if you insert new document with custom id, then delete this document, then try to add new document with this id again you will get error:

```Item with same id already exists```

```
const document = { id: 'test', text: 'Hello' }
DB.insert(document);
DB.removeOne({ id: document.id })
DB.insert(document); // <- error
```

https://github.com/maxnowack/signaldb/blob/6eed2e2c916a062b6c03be59dbbd6e8cdd4f018e/packages/signaldb/src/Collection/index.ts#L96

Looks like when removing item the idIndex isn't cleared for this id, but with custom identifiers i think this should happen cause in my scenario i use some id from other source to use it as id in signaldb and cant remove/add this id twice, actually after page refresh idIndex doesn't contain any index and this is more like bug.
 
signaldb 0.8.6

**Comment by maxnowack** [comment]
Good catch! Thanks also for your pull request. I took your commit, changed the commit message to match this repository's style, and added a new test case in #610.

**Comment by maxnowack** [comment]
released in `v0.8.7` 🙂
